tests/conftest.py

```diff
@@ -25,26 +50,391 @@ def config(kube_ns):
     """
     cfg = Config()
     cfg.KubeSpawner.namespace = kube_ns
+    cfg.KubeSpawner.cmd = ["jupyterhub-singleuser"]
     return cfg
 
 
+@pytest.fixture
+def app(config, tmpdir):
+    tmpdir.chdir()
+    app = JupyterHub(config=config)
+    return app
+
+
+@pytest.fixture(scope="session")
+def ssl_app(tmpdir_factory):
+    tmpdir = tmpdir_factory.mktemp("ssl")
```

Comment:
Docstring explaining that we are just creating the certificates, not a fully initialized JupyterHub instance